### DISCUSSION TITLE
Support for Airthings Wave radon detector with ESP32-BLE

### DISCUSSION THREAD

**Discussion by qingz2004** [arendst/Tasmota#14950] [Q&A] [answered]
Here is the detector.
https://www.airthings.com/en/wave-radon
Somebody wrote a sketch to build a MQTT bridge with a ESP32.
https://github.com/sabeechen/AirthingsMQTT

Is it possible to have it build in to Tasmota32-bluetooth?
Or at least read and decode the messages from the detector?

**Comment by Jason2866** [comment, marked as answer]
Can be done with Berry. A example for https://github.com/tony-fav/tasmota-blerry/tree/dev

**Reply by qingz2004** [reply]
Got it. Thanks!
Actually I only need the BLEOp1 command to read the data from the device. And it worked out perfect.

**Reply by trlafleur** [reply]
have you published your Airthing Tasmota code??

thanks

**Comment by SL06** [comment]
I know that this post is old but I hope to find someone with share interest. I just bought a Airthings Wave Radon.  
I can connect to the Airthings and read the datas via bluetooth.

private let serviceUUID = CBUUID(string: "B42E4A8E-ADE7-11E4-89D3-123B93F75CBA")
private let airthingsServiceUUID = CBUUID(string: "B42E4A8E-ADE7-11E4-89D3-123B93F75CBA")
private let sensorServiceUUID = CBUUID(string: "B42E4A8E-ADE7-11E4-89D3-123B93F75CBA")

private let sensorCharacteristicUUID = CBUUID(string: "B42E4DCC-ADE7-11E4-89D3-123B93F75CBA")

Data (hex): 014d40003300a0003507ffffffffffff0000ffff
Bytes 0-1: 014d = 19713 
Bytes 1-2: 4d40 = 16461
Bytes 2-3: 4000 = 64 
Bytes 3-4: 0033 = 13056
Bytes 4-5: 3300 = 51   =>  actual radon - OK
Bytes 5-6: 00a0 = 40960 
Bytes 6-7: a000 = 160   =>  long term radon ?  
Bytes 7-8: 0035 = 13568 
Bytes 8-9: 3507 = 1845 (normal)   = Temperature 18.5C ( /100) - OK
Bytes 9-10: 07ff = 65287 
Bytes 10-11: ffff = 65535  
Bytes 11-12: ffff = 65535 
Bytes 12-13: ffff = 65535 
Bytes 13-14: ffff = 65535 
Bytes 14-15: ffff = 65535 
Bytes 15-16: ff00 = 255 
Bytes 16-17: 0000 = 0 
Bytes 17-18: 00ff = 65280 
Bytes 18-19: ffff = 65535 =>  batterie ? 100%

I have found temperature and radon but struggling to identify the humidity (actual level 38.5)
 byte 0-1 seems to change with humidity  19713/500 = 39.426 but it does not corresponds exactly.  On an Airthing Wave Plus device, the calculation was values divide per 100.  I dont see any data close to 3850.    Did someone solve this mystery?

**Comment by qingz2004** [comment]
Byte 	Name 	Type 	Notes
0 	version 	uint8 	sensor version, always 0x01
1 	humidity 	uint8 	relative humidty = value/2 (percent)
2-3 	-- 	-- 	unknown, varies
4-5 	radon_sta 	uint16 	radon short-term average (Bq/m3)
6-7 	radon_lta 	uint16 	radon long-term average (Bq/m3)
8-9 	temperature 	uint16 	temp = val/100 (celsius)

**Comment by SL06** [comment]
Thanks for sharing the sensor data format.  
Problem solved .  

Regards.
 

Sylvain Lareau 
***@***.*** ***@***.***>il.com ***@***.***>
(819) 669-4962
